data/data/bootstrap/baremetal/files/etc/NetworkManager/system-connections/nmconnection.template

```diff
@@ -0,0 +1,15 @@
+{{if .PlatformData.BareMetal.ExternalStaticIP}}
+[connection]
+type=ethernet
+interface-name={{.PlatformData.BareMetal.ProvisioningInterfaceMAC}}
```

Comment:
I don't think this is what we're looking for. For a start, we're trying to configure the External interface, and this variable is for the Provisioning interface,
But also I don't know that you can just stick a MAC address as an interface name in NetworkManager? Nothing in https://networkmanager.dev/docs/api/latest/settings-connection.html suggests that to me.

In #5698 passing the MAC only worked because there is code that runs in the script that looks up the interface name for that MAC using NetworkManager https://github.com/openshift/installer/pull/5698/files#diff-6f24f49f3938f9f97f5429b1280c21ccb462b377705ad4f9a7f526316caa2784R31. What we have here is a much harder problem because we need networking to be set up in the initramfs, before we even get to run any code.

It looks from https://networkmanager.dev/docs/api/latest/NetworkManager.conf.html#connection-sections that there _might_ be a way to set in the NetworkManager config which device a connection gets mapped to when interface-name doesn't appear It seems like either:

```
[connection-external]
match-device=mac:<external mac>
```

or

```
[device-external]
match-device=mac:<external mac>
allowed-connections=uuid:<connection UUID>

[device-provisioning]
match-device=mac:<provisioning mac>
allowed-connections=except:uuid:<connection UUID>
```

might possibly be the magic incantation.

Reply:
> What we have here is a much harder problem because we need networking to be set up in the initramfs, before we even get to run any code.

@zaneb I don't think that's the case on the bootstrap VM, unlike the other hosts we're not pulling ignition in the ramdisk via the network, because it's injected via the fw-cfg qemu interface instead.

I agree we need to use the `ExternalMACAddress` though, but I think we can just use `mac-address` in the `[ethernet]` section to specify the device, e.g similar to [these docs](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/configuring_and_managing_networking/assembly_manually-creating-networkmanager-profiles-in-key-file-format_configuring-and-managing-networking) (not tested this myself though)

Reply:
> I agree we need to use the ExternalMACAddress though, but I think we can just use mac-address in the [ethernet] section to specify the device, e.g similar to [these docs](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/configuring_and_managing_networking/assembly_manually-creating-networkmanager-profiles-in-key-file-format_configuring-and-managing-networking) (not tested this myself though)

Just added these changes

Reply:
Ah yep, that looks correct, here is the doc: https://networkmanager.dev/docs/api/latest/settings-802-3-ethernet.html

> If specified, this connection will only apply to the Ethernet device  whose permanent MAC address matches. This property does not change the  MAC address of the device (i.e. MAC spoofing).

That's exactly what we want.